util/chplenv/chpl_compiler.py

```diff
@@ -197,6 +198,12 @@ def get(flag='host'):
                 compiler_val = 'clang'
             else:
                 compiler_val = 'gnu'
+        elif locale_model_val == 'gpu':
+            if find_executable('clang'):
+                compiler_val = 'clang'
+            else:
+                error("clang not found. The 'gpu' locale model is supported "
+                      "with clang only.")
```

Comment:
This has the effect of requiring that `chpl` be built with clang (i.e. the compiler source .cpp files are compiled with clang). That seem surprising to me. Is that expected? Anyway this kind of check needs to go in to validate_compiler.

Reply:
I wasn't thinking about requiring that here, and intended this as something that can give an early error message if you are using GPU locale model without clang.

Reply:
Do you mean "using the GPU locale model without CHPL_TARGET_COMPILER=llvm" ?

Reply:
Don't we need `clang`? Gotta say that the difference between the two in this context is not crystal clear to me.